### PR TITLE
UX: set notification popup title differently on test builds

### PR DESCRIPTION
**Pull request by HowardBraham**
## **Description**

On a standard build, the notification popup window title will be just `MetaMask`

For E2E tests, it is necessary to set it to something different, so it will be `MetaMask Dialog`

## **Related issues**

Followup to #21680

## **Manual testing steps**

1. On a standard build, go to https://metamask.github.io/test-dapp/ and prompt a popup
2. Confirm that the window's title is just `MetaMask`
3. Confirm that the E2E tests still work

### **Standard Build**

image

### **Test Build**

image

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: test/e2e/tests/ppom-blockaid-alert-metrics.spec.js [hunk 1]
```diff
@@ -282,35 +283,21 @@ describe('Confirmation Security Alert - Blockaid @no-mmi', function () {
         await driver.clickElement('#maliciousApprovalButton');
 
         // Wait for confirmation pop-up
-        let windowHandles = await driver.waitUntilXWindowHandles(3);
-        await driver.switchToWindowWithTitle(
-          WINDOW_TITLES.Notification,
-          windowHandles,
-        );
+        await switchToNotificationWindow(driver, 3);
```
Comment (FrederikBolding) [inline test/e2e/tests/ppom-blockaid-alert-metrics.spec.js:286]: This E2E is currently disabled. Could we extract these fixes to a separate PR?

**Review by eriknson — DISMISSED** [review]

File: test/e2e/tests/ppom-blockaid-alert-metrics.spec.js (hunk 1 above)
Comment (HowardBraham) [reply test/e2e/tests/ppom-blockaid-alert-metrics.spec.js]: Oh, hah, didn't notice the `it.skip`, I guess it happened after I started!  Reverted that file for now.

**Review by FrederikBolding — APPROVED** [review]

**Review by eriknson — APPROVED** [review]
Sweet!
